src/services/completions.ts

```diff
@@ -826,6 +830,174 @@ function jsdocCompletionInfo(entries: CompletionEntry[]): CompletionInfo {
     return { isGlobalCompletion: false, isMemberCompletion: false, isNewIdentifierLocation: false, entries };
 }
 
+function getJSDocParameterCompletions(
+    sourceFile: SourceFile,
+    position: number,
+    preferences: UserPreferences,
+    options: CompilerOptions,
+    tagNameOnly: boolean): CompletionEntry[] {
+    const currentToken = getTokenAtPosition(sourceFile, position);
+    if (!isJSDocTag(currentToken) && !isJSDoc(currentToken)) {
+        return [];
+    }
+    const jsDoc = isJSDoc(currentToken) ? currentToken : currentToken.parent;
+    if (!isJSDoc(jsDoc)) {
+        return [];
+    }
+    const func = jsDoc.parent;
+    if (!isFunctionLike(func)) {
+        return [];
+    }
+
+    const isJs = isSourceFileJS(sourceFile);
+    const isSnippet = preferences.includeCompletionsWithSnippetText || undefined;
+    const paramTagCount = jsDoc.tags?.filter(tag => isJSDocParameterTag(tag) && tag.getEnd() <= position).length;
+    return mapDefined(func.parameters, param => {
+        if (getJSDocParameterTags(param).length) {
+            return undefined; // Parameter is already annotated.
+        }
+        if (isIdentifier(param.name)) { // Named parameter
+            const tabstopCounter = { tabstop: 1 };
+            const paramName = param.name.text;
+            let snippetText = getJSDocParamAnnotation(paramName, param.initializer, isJs, /*isObject*/ false, /*isSnippet*/ true, tabstopCounter);
+            let insertText = getJSDocParamAnnotation(paramName, param.initializer, isJs, /*isObject*/ false, /*isSnippet*/ false);
+            if (tagNameOnly) { // Remove `@`
+                insertText = insertText.slice(1);
+                snippetText = snippetText.slice(1);
+            }
+            return {
+                name: insertText,
+                kind: ScriptElementKind.parameterElement,
+                sortText: SortText.LocationPriority,
+                insertText: isSnippet ? snippetText : undefined,
+                isSnippet,
+            };
+        }
+        else if (param.parent.parameters.indexOf(param) === paramTagCount) { // Destructuring parameter; do it positionally
+            const paramPath = `param${paramTagCount}`;
+            const insertTextResult =
+                generateJSDocParamTagsForDestructuring(paramPath, param.name, param.initializer, isJs, /*isSnippet*/ false);
+            const snippetTextResult =
+                generateJSDocParamTagsForDestructuring(paramPath, param.name, param.initializer, isJs, /*isSnippet*/ true);
+            for (let i = 1; i < insertTextResult.length; i++) {
+                insertTextResult[i] = `* ${insertTextResult[i]}`;
+                snippetTextResult[i] = `* ${snippetTextResult[i]}`;
+            }
+            let insertText = insertTextResult.join(getNewLineCharacter(options));
+            let snippetText = snippetTextResult.join(getNewLineCharacter(options));

```

Comment:
```suggestion
            let insertText = insertTextResult.join(getNewLineCharacter(options) + "* ");
            let snippetText = snippetTextResult.join(getNewLineCharacter(options) + "* ");
```

This also works better when `insertTextResult.length !== snippetTextResult.length` (although you might want to assert that this is never the case).